cli/config.py

```diff
@@ -114,119 +114,64 @@ def _merge_secrets_dict(secrets: SecretsConfiguration, secrets_config):
-
-        if config.get('password') is not None:
-            self.password = config['password']
+    if config.get('custom_scripts') not in [[None], None]:
+        output.custom_scripts = config['custom_scripts']
 
-        if config.get('custom_scripts') not in [[None], None]:
-            self.custom_scripts = config['custom_scripts']
+    if config.get('azure_files') not in [[None], None]:
+        output.file_shares = config['azure_files']
 
-        if config.get('azure_files') not in [[None], None]:
-            self.file_shares = config['azure_files']
+    if config.get('docker_repo') is not None:
+        output.docker_repo = config['docker_repo']
 
-        if config.get('docker_repo') is not None:
-            self.docker_repo = config['docker_repo']
-
-        if config.get('wait') is not None:
-            self.wait = config['wait']
-
-    def merge(self, uid, username, size, size_low_pri, vm_size, subnet_id, password, wait, docker_repo):
-        """
-            Reads configuration file (cluster.yaml), merges with command line parameters,
-            checks for errors with configuration
-        """
-        self._read_config_file(os.path.join(
-            aztk.utils.constants.HOME_DIRECTORY_PATH, '.aztk', 'cluster.yaml'))
-        self._read_config_file()
+    if config.get('wait') is not None:
+        wait = config['wait']
 
-        self._merge_dict(
-            dict(
-                id=uid,
-                username=username,
-                size=size,
-                size_low_pri=size_low_pri,
-                vm_size=vm_size,
-                subnet_id=subnet_id,
-                password=password,
-                wait=wait,
-                custom_scripts=None,
-                docker_repo=docker_repo
-            )
-        )
-
-        if self.uid is None:
-            raise aztk.error.AztkError(
-                "Please supply an id for the cluster with a parameter (--id)")
-
-        if self.size == 0 and self.size_low_pri == 0:
-            raise aztk.error.AztkError(
-                "Please supply a valid (greater than 0) size or size_low_pri value either in the cluster.yaml configuration file or with a parameter (--size or --size-low-pri)")
-
-        if self.vm_size is None:
-            raise aztk.error.AztkError(
-                "Please supply a vm_size in either the cluster.yaml configuration file or with a parameter (--vm-size)")
-
-        if self.wait is None:
-            raise aztk.error.AztkError(
-                "Please supply a value for wait in either the cluster.yaml configuration file or with a parameter (--wait or --no-wait)")
-
-        if self.username is not None and self.wait is False:
-            raise aztk.error.AztkError(
-                "You cannot create a user '{0}' if wait is set to false. By default, we create a user in the cluster.yaml file. Please either the configure your cluster.yaml file or set the parameter (--wait)".format(self.username))
+    return output, wait
```

Comment:
seems weird to me to return two values here.. they don't really seem to go together.

Reply:
I believe he did it this way because cluster.yaml has a "wait" parameter, while the ClusterConfiguration object does not. 

The "wait" parameter is passed to client.create_cluster() along with the ClusterConfiguration so these are returned together here.

Reply:
I could also return  `ClusterCLIParams` class here instead of a tuple

Reply:
I feel like a tuple works fine since the CLI is the only caller of this method.